dataflow/src/main/java/org/checkerframework/dataflow/busyexpression/BusyExprStore.java

```diff
@@ -0,0 +1,141 @@
+package org.checkerframework.dataflow.busyexpression;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.analysis.Store;
+import org.checkerframework.dataflow.cfg.node.BinaryOperationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.visualize.CFGVisualizer;
+import org.checkerframework.dataflow.expression.JavaExpression;
+import org.checkerframework.javacutil.BugInCF;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.StringJoiner;
+
+/** A busy expression store contains a set of busy expressions represented by nodes. */
+public class BusyExprStore implements Store<BusyExprStore> {
+
+    /** A set of busy expression abstract values. */
+    private final Set<BusyExprValue> busyExprValueSet;
+
+    /**
+     * Create a new BusyExprStore.
+     *
+     * @param busyExprValueSet a set of busy expression abstract values. The parameter is captured
+     *     and that the caller should not retain an alias
+     */
+    public BusyExprStore(Set<BusyExprValue> busyExprValueSet) {
+        this.busyExprValueSet = busyExprValueSet;
+    }
+
+    /** Create a new BusyExprStore. */
+    public BusyExprStore() {
+        busyExprValueSet = new LinkedHashSet<>();
+    }
+
+    /**
+     * Kill expressions if they contain variable var.
+     *
+     * @param var a variable
+     */
+    public void killBusyExpr(Node var) {
+        busyExprValueSet.removeIf(
+                busyExprValue -> exprContainsVariable(busyExprValue.busyExpression, var));
+    }
+
+    /**
+     * Return if the expression contains variable var.
+     *
+     * @param expr the expression checked
+     * @param var the variable
+     * @return true if the expression contains the variable
+     */
+    public boolean exprContainsVariable(Node expr, Node var) {
+        if (expr instanceof BinaryOperationNode) {
+            BinaryOperationNode binaryNode = (BinaryOperationNode) expr;
+            return exprContainsVariable(binaryNode.getLeftOperand(), var)
+                    || exprContainsVariable(binaryNode.getRightOperand(), var);
+        }
+
+        return expr.equals(var);
```

Comment:
Nodes usually should not be compared with `.equals`. Add a comment why this is the right thing to do in the method documentation.

Reply:
added it.